runtime/autoload/provider/node.vim

```diff
@@ -65,31 +55,42 @@ function! provider#node#Detect() abort
     return ''
   endif
 
-  let yarn_subpath = '/node_modules/neovim/bin/cli.js'
-  let npm_subpath = '/neovim/bin/cli.js'
-
-  " `yarn global dir` is slow (> 250ms), try the default path first
-  if filereadable('$HOME/.config/yarn/global' . yarn_subpath)
-      return '$HOME/.config/yarn/global' . yarn_subpath
-  end
+  let npm_opts = {}
+  if executable('npm')
+    let npm_opts = deepcopy(s:NodeHandler)
+    let npm_opts.entry_point = '/neovim/bin/cli.js'
+    let npm_opts.job_id = jobstart('npm --loglevel silent root -g', npm_opts)
+  endif
 
-  " try both npm and yarn simultaneously
-  let yarn_opts = s:NodeHandler.new()
-  let yarn_opts.entry_point = yarn_subpath
-  let yarn_opts.job_id = jobstart(['yarn', 'global', 'dir'], yarn_opts)
-  let npm_opts = s:NodeHandler.new()
-  let npm_opts.entry_point = npm_subpath
-  let npm_opts.job_id = jobstart(['npm', '--loglevel', 'silent', 'root', '-g'], npm_opts)
+  let yarn_opts = {}
+  if executable('yarn')
+    let yarn_opts = deepcopy(s:NodeHandler)
+    let yarn_opts.entry_point = '/node_modules/neovim/bin/cli.js'
+    " `yarn global dir` is slow (> 250ms), try the default path first
+    " XXX: The following code is not portable
+    " https://github.com/yarnpkg/yarn/issues/2049#issuecomment-263183768
+    if has('unix')
+      let yarn_default_path = $HOME . '/.config/yarn/global/' . yarn_opts.entry_point
```

Comment:
I don't use yarn so I won't fix this for Windows. I'm interested only in `missing_provider` for the nodejs functional tests.

Reply:
fwiw it's technically not non-portable code, as it'll still run without erroring on non-unix but not wind up taking the fast path.